benchmark/main.go

```diff
@@ -0,0 +1,54 @@
+package main
+
+import (
+	"fmt"
+	"github.com/pokt-network/merkle"
+	"github.com/pkg/profile"
+	"unsafe"
+)
+
+func main() {
+	Proof()
+}
+
+const (
+	numOfLeaves = 100000000
+	index       = 0
+)
+
+var data [][]byte
+
+func init() {
+	genData()
+}
+
+func Root() {
+	defer profile.Start(profile.MemProfile).Stop()
+	fmt.Println(cap(merkle.GenerateRoot(data)))
+}
+
+func Proof() {
+	prof := profile.Start(profile.MemProfile)
+	p := merkle.GenerateProof(data, index)
+	prof.Stop()
+	var size int
+	for _, hash := range p.Hashes {
+		size = size + cap(hash)
+	}
+	fmt.Println(unsafe.Sizeof(p.Index) + uintptr(size))
+}
+
+func Verify() {
+	root := merkle.GenerateRoot(data)
+	genData() // resets data
+	proof := merkle.GenerateProof(data, index)
+	defer profile.Start(profile.MemProfile).Stop()
+	merkle.VerifyProof(root, []byte("foo"), proof)
+}
+
+func genData() {
+	data = nil
+	for i := 0; i < numOfLeaves; i++ {
+		data = append(data, []byte("foo"))
+	}
```

Comment:
can refactor this function to create a separate proof object to verify on the Verify Method?
```Go
  data = nil
  for I := 0; i < numOfLeaves - 1; i++ {
     data = append(data, []byte("foo")
  }
  data = append(data, mockLeaf)
```

The `mockLeaf` should be declared on the global scope and be added and should be the leaf to verify on the `Verify` Method

I think this provides a simpler way to understand the test workflow.

Reply:
Resolved with
020bbcd